### ISSUE TITLE
Missing `shapely` dependency in `setup.py` ?

### ISSUE THREAD

**Issue by EtienneDG** [geoalchemy/geoalchemy2#440]
Hello,

I installed the package from the latest commit on master (to test out MySQL support).

When trying to run my code I got the following exception : 
```
  File "/home/.../python3.10/site-packages/geoalchemy2/__init__.py", line 6, in <module>
    from geoalchemy2 import shape  # noqa
  File "/home/.../python3.10/site-packages/geoalchemy2/shape.py", line 8, in <module>
    import shapely.wkb
ModuleNotFoundError: No module named 'shapely'
```

Fixed it by `pip install shapely`, but I guess `shapely` should go in the `install_requires`, wdyt ?

**Comment by adrien-berchet** [comment]
Hi @EtienneDG 
Thank you for this report.
In order to make `GeoAlchemy2` light to install, the `Shapely` requirement is supposed to be optional, as it is only used in specific cases or when you explicitly want to use the `geoalchemy2.shape` module. Nevertheless, I agree we should at least add an 'extras'  entry in `setup.py` and improve the docs or raise an exception with a more explanatory message.

**Comment by EtienneDG** [comment]
Thanks for the quick answer.
To be honest I didn't see [this page](https://geoalchemy-2.readthedocs.io/en/latest/shape.html) until after submitting this issue, so I think that the doc is already enough.

However, I still had a doubt and I tested again. I think there is still something that might be wrong.

First test in a brand new and empty 3.10 virtualenv :
```
$ pip install GeoAlchemy2
Collecting GeoAlchemy2
  Using cached GeoAlchemy2-0.13.2-py3-none-any.whl (43 kB)
Collecting SQLAlchemy>=1.4
  Using cached SQLAlchemy-2.0.9-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (2.7 MB)
Collecting packaging
  Using cached packaging-23.1-py3-none-any.whl (48 kB)
Collecting greenlet!=0.4.17
  Using cached greenlet-2.0.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (613 kB)
Collecting typing-extensions>=4.2.0
  Using cached typing_extensions-4.5.0-py3-none-any.whl (27 kB)
Installing collected packages: typing-extensions, packaging, greenlet, SQLAlchemy, GeoAlchemy2
Successfully installed GeoAlchemy2-0.13.2 SQLAlchemy-2.0.9 greenlet-2.0.2 packaging-23.1 typing-extensions-4.5.0

$  python
Python 3.10.7 (main, Mar 10 2023, 10:47:39) [GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import geoalchemy2
>>> 
```

**Comment by adrien-berchet** [comment]
Indeed, my bad :facepalm: 
I opened a PR to fix this.

**Comment by adrien-berchet** [comment]
@EtienneDG I merged #441 , I hope it will be better now.
